Add a larger “family–sized” water heater

db48x opened the pull request:
#### Summary
Content "Add a larger “family–sized” water heater"

#### Purpose of change

Prior to the cataclysm it was common for larger households to have installed larger water heaters so that everybody could take a hot shower every night. Commercial establishments frequently required larger water heaters in order to serve customers reliably as well. Mapgen doesn’t reflect these facts well.

#### Describe the solution

Adds one larger size of water heater, with a 55–gallon capacity.

#### Describe alternatives you've considered

75 and 100–gallon water heaters are not uncommon, but that would require adding more tank objects.

#### Testing

`make check` for syntax, and I wandered around a world verifying that both sizes were available.

#### Additional context

![Screenshot 2023-09-29 at 17-08-32 7e7928b1-8b4f-472a-8ee8-b48eb1e7932a pdf](https://github.com/CleverRaven/Cataclysm-DDA/assets/228849/631c239f-7c65-43e9-a499-ad1fe65a2086)

db48x committed:
Add a larger “family–sized” water heater

This one has a 55–gallon tank; naturally deconstructing it yields a
`55gal-drum` of the same capacity. For consistency, the stats are
taken from a Rheem model XG55T06EC50UO water heater
<https://images.thdstatic.com/catalog/pdfImages/7e/7e7928b1-8b4f-472a-8ee8-b48eb1e7932a.pdf>.

db48x committed:
add sacrificial anode rods to water heaters to prevent corrosion

Anode rods for water heaters are typically aluminum or magnesium, as
these metals have good electronegativity and don’t cost very
much. CDDA doesn’t have magnesium metal chunks or ingots, but it does
have aluminum, so it is easiest to always use aluminum.

db48x committed:
update a wide variety of buildings to use the larger water heaters

For some buildings I left the existing small water heater as is, for a
few (such as the landromat) I changed it to a larger one, but for most
I left it to chance. I didn’t add water heaters to any building that
doesn’t already have one.

db48x committed:
lint the json

GuardianDll commented on data/json/furniture_and_terrain/furniture-plumbing.json:
```diff
@@ -129,7 +129,42 @@
         { "item": "scrap_copper", "count": [ 0, 2 ] },
-        { "item": "water_faucet", "count": [ 0, 1 ] }
+        { "item": "water_faucet", "count": [ 0, 1 ] },
+        { "item": "material_aluminium_ingot", "count": 2 }
```
why a whole ingot? i doubt water heater should drop a big solid blocks of aluminum, more like a lot of aluminum scrap

db48x replied:
The name is misleading; these “ingots” are 250g each.

GuardianDll replied:
"ingot" is not about the shape, it's about the form of an item

db48x replied:
Hmm. Well, the anode rod is in fact a rod longer than it is wide. Usually masses around a kilo, and is often divided into four segments…

db48x replied:
You can do a google image search to see pictures if you want.

GuardianDll replied:
ok, then it require some tweaks for aluminum to have more components, but it's out of scope of this PR

GuardianDll commented on data/json/furniture_and_terrain/furniture-plumbing.json:
```diff
@@ -129,7 +129,42 @@
+    "id": "f_water_heater_family",
+    "name": "family–sized water heater",
+    "looks_like": "f_water_heater",
+    "description": "An insulated metal tank with a small fire used to maintain near-boiling temperatures.  There's no longer any way to power it, but any water left in the large tank from before the power went out should be pretty clean, and it could be used to store large quantities of other liquids too.",
```
> with a small fire

Should it imply it is a gas powered heater? i don't think they are popular, also player won't be able to use one on their base, right

db48x replied:
They are very popular in the north–east, because they easily have 10× the heating capacity of electric (literally 45kW vs 4.5kW). Also, the existing water heater has the same description, and it isn’t tagged as an electric appliance either.

GuardianDll commented on data/json/mapgen_palettes/military/mil_base_palette.json:
```diff
@@ -95,7 +95,7 @@
       "C": "f_crate_c",
       "L": "f_shower",
-      "6": "f_water_heater",
+      "6": [ "f_water_heater", "f_water_heater_family" ],
```
i guess military facility can benefit from even heavier heaters, and shouldn't alternate between them

db48x replied:
That’s reasonable.

db48x committed:
make the military base use the larger water heaters

It looks like the water heaters are mostly in the barracks, infirmary,
etc, so it makes sense that they would use large ones. If 100–gallon
commercial water heaters are ever added to the game, this would be a
great place to use them.

GuardianDll commented on data/json/mapgen_palettes/aircraft_carrier_palette.json:
```diff
@@ -78,7 +78,7 @@
       "S": "f_shower",
       "s": "f_sofa",
-      "V": "f_water_heater",
+      "V": [ "f_water_heater", "f_water_heater_family" ],
```
```suggestion
      "V": "f_water_heater_family",
```
Same as mil base

db48x committed:
make the carrier only use the larger water heaters as well

Co-authored-by: Anton Simakov <67688115+GuardianDll@users.noreply.github.com>

db48x commented:
I accepted your suggestion for water heater sizes on the carrier map, but I am not entirely sure that a modern carrier would have any water heaters at all. I’ve never encountered it in the game and I haven’t looked at the map, but it seems to me that a nuclear carrier would use the reactor to heat water and then pipe it around the ship instead of making electricity that is then converted back into heat in a water heater.

But if the map has water heaters, we might as well make them large ones :)
